martin/src/pg/query_tables.rs

```diff
@@ -256,16 +265,24 @@ SELECT ST_Transform(
             4326
         ) AS bounds
 FROM {schema}.{table};
-                "#
-        )
-    };
+                "#),
+                &[]
+            ).await
+        };
 
-    Ok(pool
-        .get()
-        .await?
-        .query_one(&sql, &[])
-        .await
-        .map_err(|e| PostgresError(e, "querying table bounds"))?
-        .get::<_, Option<ewkb::Polygon>>("bounds")
-        .and_then(|p| polygon_to_bbox(&p)))
+        if let Some(bounds) = query
+            .map_err(|e| PostgresError(e, "querying table bounds"))?
+            .get::<_, Option<ewkb::Polygon>>("bounds")
+        {
+            return Ok(polygon_to_bbox(&bounds));
+        }
+        if is_quick {
+            // ST_EstimatedExtent failed probably because there is no index or statistics or if it's a view
+            // This can only happen once if we are in quick mode
+            is_quick = false;
+            warn!("ST_EstimatedExtent on {schema}.{table}.{geometry_column} failed, trying slower method to compute bounds");
```

Comment:
The user with a big and special naming table and a `auto_bounds: quick` would still get a slow result. Is this what they want? 
But it would be rare.

Reply:
exactly, this is not a common case.  My bigger concern is that we currently do not cancel a query after 5 seconds - something we may need to dig deeper into postgres_tokio crate